File: 2020/6.jl
include("utils.jl")
using .AoCUtils: readInputWithBlank

function loadInput()
    filename = "input_6.txt"
    lines = readInputWithBlank(filename)
    answers = Vector{Vector{String}}()
    temp = String[]
    for l in lines
        if l == ""
            push!(answers, temp)
            temp = String[]
        else
            push!(temp, l)
        end
    end
    push!(answers, temp)
    
    return answers
end

function part1(answers::Vector{Vector{String}})
    counts = 0
    for g in answers
        counts += length(Set([item for a in g for item in collect(a)]))
    end
    println("🎄 Part 1: $(counts)")
end

function part2(answers::Vector{Vector{String}})
    counts = 0
    for g in answers
        counts += length(reduce(intersect, g))
    end
    println("🎄🎅 Part 2: $(counts)")
end

println("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")
println("⛄        Day 6         ⛄")
println("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")

inputs = loadInput()
@time part1(inputs)
@time part2(inputs)

File: 2020/6.py
import time
from utils import readInputWithBlank

def loadInput():
    lines = readInputWithBlank("input_6.txt")
    answers = []
    temp = []
    for l in lines:
        if l == "":
            answers.append(temp)
            temp = []
        else:
            temp.append(l)

    return answers

def part1(answers):
    counts = 0
    for a in answers:
        temp = []
        for i in a:
            temp += list(i)
        counts += len(set(temp))
    print ("🎄 Part 1: {}".format(counts))

def part2(answers):
    counts = [len(set.intersection(*list(map(set, a)))) for a in answers]
    count = sum(counts)
    print ("🎄🎅 Part 2: {}".format(count))

print("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")
print("⛄        Day 6         ⛄")
print("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")

inputs = loadInput()

t0 = time.time()
part1(inputs)
print ("Time: {:.5f}".format(time.time()-t0))

t0 = time.time()
part2(inputs)
print ("Time: {:.5f}".format(time.time()-t0))
